### PR TITLE
Allow dry-run with our role

### PR DESCRIPTION
**Pull request by hush-hush**
- Fix the role to be compatible with ansible dry-run
- Fix downgrading from agent 5 to 6 on centos
- Improve our travis tests

### CODE REVIEW

**Comment by chaosoldier** [comment]
@hush-hush 
Thanks for fixing my issue. 
When is this PR merged?

**Comment by hush-hush** [comment]
@chaosoldier I'm waiting for a review from @olivielpeau. We're currently focusing on releasing the next version of the agent (6.2.0). We will merge and release the ansible role as soon as possible,

**Comment by chaosoldier** [comment]
@hush-hush
Thank you for replying. 

> We will merge and release the ansible role as soon as possible,

I understood!

File: README.md
Comment (olivielpeau) [inline README.md]: nit: `ansible 2.4 and up`?

File: tasks/pkg-debian.yml
Comment (olivielpeau) [inline tasks/pkg-debian.yml]: why is this needed? Shouldn't the package be always in the apt repo?

File: tasks/pkg-debian.yml
Comment (olivielpeau) [inline tasks/pkg-debian.yml:30]: same question

File: tasks/pkg-debian.yml
Comment (olivielpeau) [inline tasks/pkg-debian.yml:37]: same question

File: README.md
Comment (olivielpeau) [reply README.md]: you should also update the description of the `datadog_agent_allow_downgrade` variable in the `Role Variables` section of the readme

**Review by olivielpeau — COMMENTED** [review]
One general comment: according to https://docs.ansible.com/ansible/devel/user_guide/playbooks_checkmode.html#information-about-check-mode-in-variables, the variable `ansible_check_mode` is only available on versions >=2.1 of ansible. What happens on earlier versions of ansible?

Other than that and the inline nits, LGTM! The tests are very nice 👌

**Comment by hush-hush** [comment]
@olivielpeau ansible 2.1 entered end of life last year (2017-06-01). I think we can no longer support it ?

File: tasks/pkg-debian.yml
Comment (hush-hush) [reply tasks/pkg-debian.yml]: if no ansible job using apt was ever run, `--check` will fail because `python-apt` is not install.

```
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "python-apt must be installed to use check mode. If run normally this module can auto-install it."}
```

Adding the check allow users to check their playbook without installing python-apt.

**Review by olivielpeau — APPROVED** [review]
LGTM!

I'm fine with dropping support of <2.1 ansible versions, but let's not forget to update the declared minimum supported version of ansible here: https://github.com/DataDog/ansible-datadog/blob/2.0.3/meta/main.yml#L6 :)

**Pull request merged**
